FEXCore/Source/Interface/Core/JIT/JITClass.h

```diff
@@ -483,23 +483,31 @@ class Arm64JITCore final : public CPUBackend, public Arm64Emitter {
 
   template<ARMEmitter::IsLabel T>
   void adr_OrRestart(ARMEmitter::Register rd, T* Label) {
+    if (RequiresFarARM64Jumps) {
+      if (LongAddressGen(rd, Label) == ARMEmitter::BranchEncodeSucceeded::Failure) {
```

Comment:
Oh heck, I actually just looked at the implementation of `LongAddressGen` in the CodeEmitter and it only uses adr or adrp+add.
We actually need a new label type that is guaranteed to emit 3-nops to be able to encode the 48-bit movz+movk+movk in the worst case. (Long Long address gen? It's a good advertising campaign reference).